sdk/cargo-build-bpf/src/main.rs

```diff
@@ -150,25 +156,15 @@ fn install_if_missing(
             .map_err(|err| err.to_string())?;
         fs::remove_file(file).map_err(|err| err.to_string())?;
     }
-    // Make a symbolyc link source_path -> target_path in the
+    // Make a symbolic link source_path -> target_path in the
     // sdk/bpf/dependencies directory if no valid link found.
     let source_base = config.bpf_sdk.join("dependencies");
     if !source_base.exists() {
         fs::create_dir_all(&source_base).map_err(|err| err.to_string())?;
     }
     let source_path = source_base.join(package);
     // Check whether the correct symbolic link exists.
-    let invalid_link = if let Ok(link_target) = source_path.read_link() {
-        if link_target != target_path {
-            fs::remove_file(&source_path).map_err(|err| err.to_string())?;
-            true
-        } else {
-            false
-        }
-    } else {
-        true
-    };
-    if invalid_link {
+    if source_path.read_link().is_err() {
```

Comment:
the fs::remove_file might be needed here, if the link targets an older version of bpf-tools.

Reply:
Agreed, though it might be better to have a check instead to see if an old version of bpf-tools is stored at 'source_path', rather than check if the resolved link path has changed. The check could be done by parsing 'version.md' under bpf-tools.

Though, that being said, this PR is just a quick-fix for the time being since NixOS users are unable to run cargo-build-bpf right now.

Reply:
We're ok to merge this @dmakarov?  The PR author won't have the permissions so we need to in this case